packages/mobile/src/components/core/Hyperlink.tsx

```diff
@@ -68,47 +68,55 @@ export const Hyperlink = (props: HyperlinkProps) => {
    * android doesn't return the correct layout for nested text elements
    * */
   useEffect(() => {
-    Object.entries(linkRefs.current).forEach(([index, ref]) => {
-      ref.measureInWindow((x, y, width, height) => {
-        setLinkLayouts((linkLayouts) => ({
-          ...linkLayouts,
-          [index]: { x, y, width, height }
-        }))
-      })
+    Object.entries(links).forEach(([index]) => {
+      const linkRef = linkRefs[index]
+      if (linkRef) {
+        linkRef.measureInWindow((x, y, width, height) => {
+          setLinkLayouts((linkLayouts) => ({
+            ...linkLayouts,
+            [index]: { x, y, width, height }
+          }))
+        })
+      }
     })
 
     if (linkContainerRef.current) {
       linkContainerRef.current.measureInWindow((x, y, width, height) =>
         setLinkContainerLayout({ x, y, width, height })
       )
     }
-  }, [linkRefs, linkContainerRef])
+  }, [links, linkRefs, linkContainerRef])
 
   const handlePress = useOnOpenLink(source)
 
   const renderLink = useCallback(
     (text, match, index) => (
       <View
         onLayout={(e) => {
-          setLinks({
+          setLinks((links) => ({
```

Comment:
callback the onLayout do anything?

Reply:
I think fine in this case because basically a leaf component that is rerendering anyway